### PR TITLE
Added documentation on how to provide access token

### LINKED ISSUE #15466: Integrate Custom Views with Ledger API authorization

### PR DESCRIPTION
**Pull request by chunlokling-da**
fixes #15466

### CODE REVIEW

File: docs/source/app-dev/custom-views/index.rst [hunk 1]
```diff
@@ -326,3 +326,39 @@ A default dedicated dispatcher for blocking operations (e.g. db operation) is co
...
+
+.. code-block:: scala
+
+    val clientSettings = GrpcClientSettings
+      .connectToServiceAt(host, port.value)
+      .withCallCredentials(new LedgerCallCredentials(accessToken))
+    val source = BatchSource.events(clientSettings)
+    val control = Projection.project(source, exercisedEvents)(f)
+
+Provide a newly retrieved access token when the existing one expired
```
Comment (ray-roestenburg-da) [inline docs/source/app-dev/custom-views/index.rst:349]:
```suggestion
Provide a newly retrieved access token when the existing one has expired
```

File: docs/source/app-dev/custom-views/index.rst [hunk 2]
```diff
@@ -326,3 +326,39 @@ A default dedicated dispatcher for blocking operations (e.g. db operation) is co
...
+--------------------------------------------------------------------
```
Comment (ray-roestenburg-da) [inline docs/source/app-dev/custom-views/index.rst:350]:
```suggestion
------------------------------------------------------------------------
```

File: docs/source/app-dev/custom-views/index.rst [hunk 3]
```diff
@@ -326,3 +326,39 @@ A default dedicated dispatcher for blocking operations (e.g. db operation) is co
...
+Ledger API Authorization
+========================
+
+The client must provide an access token when authorization is required by the Ledger.
+For details of ledger authorization, please refer to `Ledger Authorization documentation <https://docs.daml.com/app-dev/authorization.html>`_.
+
+Provide access token to custom-view library
+-------------------------------------------
+
+Applications can provide an access token when setting up the client.
```
Comment (ray-roestenburg-da) [inline docs/source/app-dev/custom-views/index.rst:339]:
```suggestion
Applications can provide an access token when setting up the client. The example below shows how to set `LedgerCallCredentials` on the `GrpcClientSettings`.
```

File: docs/source/app-dev/custom-views/index.rst (hunk 1 above)
Comment (ray-roestenburg-da) [reply docs/source/app-dev/custom-views/index.rst:349]: Please provide a Java example instead.

File: docs/source/app-dev/custom-views/index.rst [hunk 4]
```diff
@@ -326,3 +326,39 @@ A default dedicated dispatcher for blocking operations (e.g. db operation) is co
...
+    val control = Projection.project(source, exercisedEvents)(f)
+
+Provide a newly retrieved access token when the existing one expired
+--------------------------------------------------------------------
+
+When an access token is expired, an application can retrieve a new access token with the stored refresh token.
+For details on the refresh token, please refer to `Ledger auth-middleware documentation <https://docs.daml.com/tools/auth-middleware/index.html#refresh-access-token>`_.
+With the new access token, an application can cancel the running projection and re-create a new one using the new token.
+
+.. code-block:: scala
```
Comment (ray-roestenburg-da) [inline docs/source/app-dev/custom-views/index.rst:356]: Please provide a Java example instead.

Comment (chunlokling-da) [reply docs/source/app-dev/custom-views/index.rst:356]: added

File: docs/source/app-dev/custom-views/index.rst (hunk 2 above)
Comment (chunlokling-da) [reply docs/source/app-dev/custom-views/index.rst:350]: ah right. it is too short. added more dashes

**Review by ray-roestenburg-da — APPROVED** [review]

**Pull request merged**